plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/completion/Completions.kt

```diff
@@ -24,6 +24,9 @@ internal object Completions {
                 complete(factory.classifierContributor(0), positionContext)
                 complete(factory.keywordContributor(1), positionContext)
                 complete(factory.packageCompletionContributor(2), positionContext)
+                // For `val` and `fun` completion. For example, with `val i<caret>`, the fake file contains `val iX.f`. Hence a
```

Comment:
I am not sure that this commentary is needed here (at least the part about the `iX.f`)

Reply:
Hmm, I think this comment is useful. Previously Ilya was asking why we need to complete unresolved name in a type reference. So I added a comment explaining how the placeholder causes the parser to treat the current caret position as a type reference. If the placeholder text does not contain `.`, we would not need this contributor here.